### PR TITLE
Expose more stuff in api

### PR DESCRIPTION
**Pull request by rwols**
See https://github.com/sublimelsp/LSP-elixir/pull/9#discussion_r499389741

### CODE REVIEW

File: plugin/__init__.py [hunk 1]
```diff
@@ -8,19 +10,25 @@
...
 # This is the public API for LSP-* packages
 __all__ = [
     '__version__',
     'AbstractPlugin',
     'ClientConfig',
+    'css',
+    'filename_to_uri',
     'LanguageConfig',
     'Notification',
+    'Promise',
```
Comment (rchl) [inline plugin/__init__.py:26]: Should we really do that? For `on_pre_server_command` we could just expect an optional function to be returned and internally we would use a Promise.

**Review by rchl — APPROVED** [review]

File: plugin/core/sessions.py [hunk 2]
```diff
@@ -791,8 +792,8 @@ def _template_variables(self) -> Dict[str, str]:
 
     def run_command(self, command: Mapping[str, Any]) -> Promise:
         if self._plugin:
-            promise = self._plugin.on_pre_server_command(command)
-            if promise:
+            promise, callback = Promise.packaged_task()
```
Comment (predragnikolic) [inline plugin/core/sessions.py:795]: I find the usage of `Promise.packaged_task` a bit confusing. 

If you want run a callback when the `on_pre_server_command` is done, why not write `self._plugin.on_pre_server_command(command).then(callback)`?

If we don't have the `Promise.packaged_task` I would find the code easier to understand.

Comment (rchl) [reply plugin/core/sessions.py:795]: > If you want run a callback when the `on_pre_server_command` is done, why not write `self._plugin.on_pre_server_command(command).then(callback)`?

`.on_pre_server_command(command)` doesn't return a Promise so you can call `.then` on it.

Comment (predragnikolic) [reply plugin/core/sessions.py:795]: Oh, I tough it was a promise based on the old code I saw:
```diff
-            promise = self._plugin.on_pre_server_command(command)
```

**Pull request merged**